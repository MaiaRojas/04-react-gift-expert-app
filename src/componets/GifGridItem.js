import React from 'react';
import PropTypes from 'prop-types'


export const GifGridItem = ({ title, url}) => { 
  return (
    <div className="card animate__fadeIn">
      <p>{ title }</p>
      <img src={url} alt={title} />
    </div>
  )
};

GifGridItem.protoTypes = {
  title: PropTypes.string.isRequired,

}