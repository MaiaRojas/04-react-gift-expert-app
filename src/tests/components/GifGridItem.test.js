import { shallow }from 'enzyme';
import React from 'react';
import { GifGridItem } from '../../componets/GifGridItem'

describe('Pruebas en <GrifGridItem />', () => {
  const title ='A title';
  const url = 'https://localhost/algo.jpg'
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test('should coindidir con snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should have a text with title', () => {
    const text = wrapper.find('p')
    expect( text.text().trim() ).toBe( title);
  });


  test('should have a image equal to url and alt to props', () => {
    const img = wrapper.find('img');

    expect( img.props().src ).toBe( url );
    expect( img.props().alt ).toBe( title );
  });

  test('should have class animate_fadeIn', () => {
    const div = wrapper.find('div');
  
    expect( div.prop('className').includes('animate__fadeIn') ).toBe( true );
  });
  
  
})