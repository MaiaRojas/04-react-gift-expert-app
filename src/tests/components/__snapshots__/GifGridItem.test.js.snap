// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GrifGridItem /> should coindidir con snapshot 1`] = `
<div
  className="card animate__fadeIn"
>
  <p>
    A title
  </p>
  <img
    alt="A title"
    src="https://localhost/algo.jpg"
  />
</div>
`;
